packages/flutter_tools/test/flutter_manifest_test.dart

```diff
@@ -0,0 +1,376 @@
+      expect(flutterManifest.fonts, isEmpty);
+      expect(flutterManifest.assets, isEmpty);
+    });
+
+    test('usesMaterialDesign flag is set', () async {
+      final String manifest = '''
+name: test
+dependencies:
+  flutter:
+    sdk: flutter
+flutter:
+  uses-material-design: true
+''';
+      final FlutterManifest flutterManifest = await FlutterManifest.createManifestFromString(manifest);
+      expect(flutterManifest, isNotNull);
+      expect(flutterManifest.appName, 'test');
+      expect(flutterManifest.usesMaterialDesign, true);
+      expect(flutterManifest.fontsDescriptor, isEmpty);
+      expect(flutterManifest.fonts, isEmpty);
+      expect(flutterManifest.assets, isEmpty);
+    });
+
+    test('has two assets', () async {
+      final String manifest = '''
+name: test
+dependencies:
+  flutter:
+    sdk: flutter
+flutter:
+  uses-material-design: true
+  assets:
+    - a/foo
+    - a/bar
+''';
+      final FlutterManifest flutterManifest = await FlutterManifest.createManifestFromString(manifest);
+      expect(flutterManifest, isNotNull);
+      expect(flutterManifest.appName, 'test');
+      expect(flutterManifest.usesMaterialDesign, true);
+      expect(flutterManifest.fontsDescriptor, isEmpty);
+      expect(flutterManifest.fonts, isEmpty);
+
+      final List<String> assets = flutterManifest.assets;
+      expect(assets.length, 2);
+      expect(assets, contains('a/foo'));
+      expect(assets, contains('a/bar'));
+    });
+
+    test('has one font family with one asset', () async {
+      final String manifest = '''
+name: test
+dependencies:
+  flutter:
+    sdk: flutter
+flutter:
+  uses-material-design: true
+  fonts:
+    - family: foo
+      fonts:
+        - asset: a/bar
+''';
+      final FlutterManifest flutterManifest = await FlutterManifest.createManifestFromString(manifest);
+      expect(flutterManifest, isNotNull);
+      expect(flutterManifest.appName, 'test');
+      expect(flutterManifest.usesMaterialDesign, true);
+      expect(flutterManifest.assets, isEmpty);
+
+      expect(flutterManifest.fontsDescriptor.toString(), '[{fonts: [{asset: a/bar}], family: foo}]');
+      final List<Font> fonts = flutterManifest.fonts;
+      expect(fonts.length, 1);
+      final Font font = fonts[0];
+      final String fontDescriptor = '{fonts: [{asset: a/bar}], family: foo}';
+      expect(font.getDescriptor().toString(), fontDescriptor);
+      expect(font.familyName, 'foo');
+      final List<FontAsset> assets = font.fontAssets;
+      expect(assets.length, 1);
+      final FontAsset fontAsset = assets[0];
+      expect(fontAsset.asset, 'a/bar');
+      expect(fontAsset.weight, isNull);
+      expect(fontAsset.style, isNull);
+    });
+
+    test('has one font family with a simple asset and one with weight', () async {
+      final String manifest = '''
+name: test
+dependencies:
+  flutter:
+    sdk: flutter
+flutter:
+  uses-material-design: true
+  fonts:
+    - family: foo
+      fonts:
+        - asset: a/bar
```

Comment:
It seems weird to allow duplicate asset names - which one takes precedence when this happens?

Reply:
They are not entirely duplicates as one has an associated weight. In general I think it could make sense in a setting with several different weights. You might want to use one asset for some weights and another asset for other weights.